### PR TITLE
T2165

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @brett-etherworks received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5d797fbedf9c9e7/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5d797fbedf9c9e7/output.txt

Total script time: 0.72 mins

Server is running on http://pr913.butyr.org/templates/basic/
Run the Test Suite: http://pr913.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5a571951908444a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/1e07734243e1a42/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/5a571951908444a/output.txt

Total script time: 2.49 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/1e07734243e1a42/output.txt

Total script time: 1.16 mins

Server is running on http://pr913.butyr.org/templates/basic/
Run the Test Suite: http://pr913.butyr.org/test/

File: templates/assets/editors/editorhelper.js [hunk 1]
```diff
@@ -76,6 +76,41 @@
...
+     * Member: contentEditable
+     *
+     * Makes a container's content editable using contenteditable
+     *
+     * @paran {TrackEvent} trackEvent: The trackEvent to update when content changes
+     * @param {DOMElement} contentContainer: the container which to listen for changes and set as editable
+     * @param {media} The current media's target element in Butter ( parent container )
+     */
+    global.EditorHelper.contentEditable = function( trackEvent, contentContainer, mediaContainer ) {
+      var media = mediaContainer.getBoundingClientRect(),
```
Comment (mjschranz) [inline templates/assets/editors/editorhelper.js]: It looks like you can get away with both this variable and the declaration of mediaContainer in the function parameters.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/f5b11b0bf4ce764/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 2

Live output at: http://botio.butyr.org:8000/b26161e0642d171/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/f5b11b0bf4ce764/output.txt

Total script time: 1.77 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/b26161e0642d171/output.txt

Total script time: 1.23 mins

Server is running on http://pr913.butyr.org/templates/basic/
Run the Test Suite: http://pr913.butyr.org/test/

File: templates/assets/editors/editorhelper.js [hunk 2]
```diff
@@ -76,6 +76,41 @@
       });
     };
 
+    /**
+     * Member: contentEditable
+     *
+     * Makes a container's content editable using contenteditable
+     *
+     * @paran {TrackEvent} trackEvent: The trackEvent to update when content changes
```
Comment (cadecairos) [inline templates/assets/editors/editorhelper.js]: paran -> param

File: templates/assets/editors/editorhelper.js [hunk 3]
```diff
@@ -76,6 +76,41 @@
...
+     * @param {DOMElement} contentContainer: the container which to listen for changes and set as editable
+     */
+    global.EditorHelper.contentEditable = function( trackEvent, contentContainer ) {
+      var newText = "";
+
+      if ( contentContainer ) {
+        contentContainer.addEventListener( "blur", function( e ) {
+          newText = contentContainer.textContent;
```
Comment (cadecairos) [inline templates/assets/editors/editorhelper.js:92]: We need to file a bug to ensure that we sanitize user input. (By we, I mean you :P)

File: templates/assets/editors/editorhelper.js [hunk 4]
```diff
@@ -76,6 +76,41 @@
...
+          trackEvent.update({
+            text: newText && newText !== "" ? newText : trackEvent.popcornOptions.text
+          });
+        }, false );
+        contentContainer.addEventListener( "keydown", function( e ) {
+          newText = contentContainer.textContent;
```
Comment (cadecairos) [inline templates/assets/editors/editorhelper.js]: is there a reason it needs to update newText on every keydown?

**Pull request closed without merging**